### PR TITLE
Project: Weather App - Jamie Cook

### PR DESCRIPTION
**Pull request by the-j-curl**
Link to my deployed project:
https://project-weatherapp.netlify.app/

My reflections on how this week's project turned out:
While I enjoyed learning about API's and JSON, I also found it tough to get started in this weeks project. I feel that I understood the fetch function and how to access the data, but knowing how to capture it and then manipulate it using functions provide to be confusing.

Things I'd like to have clarified or explained in more detail:
How to plan my JavaScript so that I know what functions are required, and how they will all link up together.

### CODE REVIEW

**Review by sarahejhej — CHANGES_REQUESTED** [review]
I'm really impressed by your project this week Jamie, I know you had a late and stressful start. You must have put a lot of effort into it this weekend, bravo! Your code is clean and structured, it's easy to read and understand. Your class and id naming seem well thought out and you use semantic HTML tags. The design is really clean and I like the cards you made for the weather forecasts! The page is also responsive and designed mobile first  👍

There seems to be duplicates of some of the files, I think you could delete all the extra files and folders. The indentation is mostly consistent, but consider changing it to 2 spaces. It's easy to change the formatting settings in VS code to auto format. And in the HTML file there were a mix of 2 and 4 spaces, please chose one and be consistent. Also remove the console.logs that were left. That's all to be done before being approved! 😄🏅

File: index.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply index.html:5]: The 4 space indentation was automatically generated by visual code. I wonder if there is a setting to change this?

File: index.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          <div class="hamburger">
+            <div></div>
+            <div></div>
+            <div></div>
+          </div>
+        </nav> 
+        <h1 id="todays-temperature" class="heading-temperature"></h1>
+        <h2 id="main-city" class="main-city"></h2>
+        <p id="main-description" class="main-description"></p>
+        <p id="sun-times" class="sun-times"></p>
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply index.html:24]: Yes good point. I have updated to main-temperature

File: index.html [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+<body>
+  
+    <section id="todays-weather" class="todays-weather">
+        <nav>
+          <div class="hamburger">
+            <div></div>
+            <div></div>
+            <div></div>
+          </div>
+        </nav>
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply index.html:20]: Again, auto indentation from VS. Not sure why. Corrected

File: style.css [hunk 4]
```diff
@@ -0,0 +1,147 @@
+body {
+    margin: 0;
+    padding: 0;
+    font-family: 'Epilogue', sans-serif;
+    background-color: #e6f3f5;
+    text-align: center;
+    font-family: 'Kumbh Sans', sans-serif;
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply style.css:7]: I was being greedy there! I think this was where I was testing different fonts.
One has now been removed. Thanks

File: style.css [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+.todays-weather {
+    height: 400px;
+    margin-bottom: 30px;
+    background-image: url(./assets/sky-day.jpg);
+    background-repeat: no-repeat;
+    background-attachment: local;
+    background-position-x: center;
+    background-position-y: 60%;
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply style.css:42]: Yes this is quite useful when you want to have more control over the background image. For example, some parts of this image show's a lot of cloud, which meant that my white text did not stand out enough.

File: style.css [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: space-between;
+    text-align: center;
+}
+
+.heading-temperature {
+    font-size: 70px;
+}
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply style.css:69]: Good spot

File: style.css [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+.five-day-forecast {
+    background: rgba(250, 251, 252, 0.9);
+    margin-bottom: 8px;
+    width: 100%;
+    display: flex;
+    flex-direction: row;
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply style.css:72]: Yes correct. Is it good practise to specify it though anyway? I'm not sure.
Maybe a question for a code coach.

**Comment by the-j-curl** [comment]
Thanks for your feedback Sara. I always appreciate other peoples thoughts and comments, and yours have been useful.
I have completed the changes and pushed them to github.

File: script.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+const containerMainDescription = document.getElementById('main-description');
+const containerSunTimes = document.getElementById('sun-times');
+const forecastContainer = document.getElementsByClassName('five-day-forecast');
+
+fetch(API_URL1) // first fetch to get todays weather
+    .then((response) => {
+        return response.json();
+    })
+    .then((todaysResults) => {
+       const city = `${todaysResults.name}`;
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply script.js:16]: Yes, I think you're right. Updated

File: script.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+       const temperature = tempConvert(todaysResults.main.temp);
+       const description = todaysResults.weather[0].description;
+       const sunrise = timeConvert(todaysResults.sys.sunrise);
+       const sunset = timeConvert(todaysResults.sys.sunset);
+       containerTodaysTemp.innerHTML += `${temperature}\u00B0`;
+       containerCurrentCity.innerHTML += city;
+       containerMainDescription.innerHTML += description.toUpperCase();
+       containerSunTimes.innerHTML += `Sunrise: ${sunrise} | Sunset: ${sunset}`;
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply script.js:24]: Removed

File: script.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                const filteredArray = forecastArray.list.filter((item) => 
+                    item.dt_txt.includes("12:00")
+                );
+                // map to create new and filtered array of weather forecast
+                const forecasts = filteredArray.map((forecast) => {
+                    const day = setDayDate(forecast.dt);
+                    const iconSrc = `http://openweathermap.org/img/wn/${forecast.weather[0].icon}@2x.png`;
+                    const temperature = tempConvert(forecast.main.temp);
+                    const feelsLike = tempConvert(forecast.main.feels_like);
+                    console.log(day, iconSrc, temperature, feelsLike);
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply script.js:42]: Removed

File: script.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+                    return {day, iconSrc, temperature, feelsLike};
+                });
+                // forEach loop to print data to HTML
+                forecasts.forEach((item, index) => {
+                    console.log(item);
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply script.js:47]: Removed

File: script.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+const timeConvert = (time) => {
+    const sun = new Date(time * 1000);
+    const sunTimeString = sun.toLocaleTimeString([], {
+        hour:"2-digit", 
+        minute:"2-digit",
+    });
+    return sunTimeString;
+};
+
+const setDayDate = (day) => {
```
[the remark this answers is not in the document]
Comment (the-j-curl) [reply script.js:68]: Yes I agree - updated

File: style.css (hunk 7 above)
[the remark this answers is not in the document]
Comment (the-j-curl) [reply style.css:72]: Cool. In that case I have removed this line

**Review by sarahejhej — APPROVED** [review]

File: script.js [hunk 13]
```diff
@@ -0,0 +1,78 @@
+const API_KEY = '2c57bc1d70d090d5c103c4167c1a5a74';
+const API_URL1 = `https://api.openweathermap.org/data/2.5/weather?q=Stockholm,Sweden&units=metric&APPID=${API_KEY}`;
+const API_URL2 = `https://api.openweathermap.org/data/2.5/forecast?q=Stockholm,Sweden&units=metric&APPID=${API_KEY}`;
```
Comment (EmOlsz) [inline script.js:3]: `API_URL_WEATHER` and `API_URL_FORECAST` as names would be better. Numbers in variable names are not very descriptive

File: script.js [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+                forecasts.forEach((item, index) => {
+                    forecastContainer[index].querySelector('.day').innerText = item.day;
+                    forecastContainer[index].querySelector('.icon').src = item.iconSrc;
+                    forecastContainer[index].querySelector('.temperature').innerText = `${item.temperature}\u00B0 / ${item.feelsLike}\u00B0`;
+                });
+             });        
+
+
+function tempConvert(number) {
+    return Math.round(number * 10) / 10;
```
Comment (EmOlsz) [inline script.js:54]: Nice job :)

File: style.css [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+
+.main-city {
+    margin-top: 5px;
+    font-size: 40px;
+}
+
+.todays-weather {
+    height: 400px;
+    margin-bottom: 30px;
+    background-image: url(./assets/sky-day.jpg);
```
Comment (EmOlsz) [inline style.css:38]: Let's try to wrap url inside quotes - some browsers might have problem with reading url without them :

```
background-image: url("./assets/sky-day.jpg");
```

File: style.css [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+    background-repeat: no-repeat;
+    background-attachment: local;
+    background-position-x: center;
+    background-position-y: 60%;
+    color: #ffffff;
```
Comment (EmOlsz) [inline style.css:43]: We can simplify this : `#fff`

**Review by EmOlsz — APPROVED** [review]
Good job Jamie, I really had hard time trying to find things in your code which could be improved :) Keep it this way!
I've added some small 4 comments, take a look on them!

**Pull request closed without merging**